internal/pkg/model/v1/gateway.go

```diff
@@ -19,23 +22,39 @@ type Gateway struct {
 	MashlingConfig interface{}
 	SchemaVersion  string
 	ErrorDetails   []gwerrors.Error
+	PingService    services.PingService
 }
 
 // Init initializes the Gateway.
-func (g *Gateway) Init() error {
+func (g *Gateway) Init(pingPort string) error {
 	log.Println("[mashling] Initializing Flogo engine...")
+
+	g.PingService = services.GetPingService()
+	g.PingService.Init(pingPort, services.PingResponse{g.Version(), g.AppVersion(), g.Description()})
+
 	return g.FlogoEngine.Init(true)
 }
 
 // Start starts the Gateway.
 func (g *Gateway) Start() error {
 	log.Println("[mashling] Starting Flogo engine...")
-	return g.FlogoEngine.Start()
+	err := g.FlogoEngine.Start()
+	if err != nil {
+		return err
+	}
+	err = g.PingService.Start()
+	if err != nil {
+		g.Stop()
+		fmt.Println(err)
+		os.Exit(1)
```

Comment:
Since err is not being handled by the caller of this script (i.e run.go), it has to be handled here OR inside pingService.Start(). To make it inline with flogoEngine.Start(), I would suggest err handling can be done inside pingService.